pkg/synapse/materialization_test.go

```diff
@@ -0,0 +1,188 @@
+package synapse
+
+import (
+	"testing"
+
+	"github.com/bruin-data/bruin/pkg/pipeline"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMaterializer_Render(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name    string
+		task    *pipeline.Asset
+		query   string
+		want    []string
+		wantErr bool
+	}{
+		{
+			name:  "no materialization, return raw query",
+			task:  &pipeline.Asset{},
+			query: "SELECT 1",
+			want:  []string{"SELECT 1"},
+		},
+		{
+			name: "materialize to a view",
+			task: &pipeline.Asset{
+				Name: "my.asset",
+				Materialization: pipeline.Materialization{
+					Type: pipeline.MaterializationTypeView,
+				},
+			},
+			query: "SELECT 1",
+			want: []string{
+				"^DROP VIEW IF EXISTS my\\.asset;$",
+				"^CREATE VIEW my\\.asset AS SELECT 1;$",
```

Comment:
here a single statement would solve it I guess, CREATE OR ALTER VIEW

Reply:
There is no `CREATE OR ALTER VIEW...` in synapse sql, despite it existing in ms sql